Invalidate FileInfo after CreateText and AppendText

hangy opened the pull request:
Similarly to `FileInfo.Create()`, `FileInfo.CreateText()` and `FileInfo.AppendText()` now invalidate the data cached inside a `FileInfo` instance, so that `FileInfo.Exists` returns the correct value when being called afterwards.

Fix #77453

hangy committed:
Invalidate FileInfo after CreateText and AppendText

adamsitnik commented on src/libraries/System.IO.FileSystem/tests/FileInfo/CreateText.cs:
```diff
@@ -0,0 +1,22 @@
+        [Fact]
+        public void FileInfoInvalidAfterCreateText()
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            string testFilePath = Path.Combine(testDir.FullName, GetTestFileName());
+            FileInfo info = new FileInfo(testFilePath);
```
```suggestion
            FileInfo info = new FileInfo(GetTestFilePath());
```

adamsitnik commented on src/libraries/System.Private.CoreLib/src/System/IO/FileInfo.cs:
```diff
@@ -200,5 +200,12 @@ public FileInfo Replace(string destinationFileName, string? destinationBackupFil
+
+        private StreamWriter CreateStreamWriter(bool append)
+        {
+            StreamWriter streamWriter = new StreamWriter(NormalizedPath);
```
we need to pass `append` argument to the `StreamWriter` ctor:

```suggestion
            StreamWriter streamWriter = new StreamWriter(NormalizedPath, append);
```

adamsitnik reviewed (commented):
@hangy thank you for your contribution! Please take a look at my comments.

hangy committed:
fix: Pass `append` to `StreamWriter` ctor

Co-authored-by: Adam Sitnik <adam.sitnik@gmail.com>

hangy committed:
reefactor: Apply suggestions from code review

Skip creation of a temporary test directory, and use inlined `GetTestFilePath()` instead.

Co-authored-by: Adam Sitnik <adam.sitnik@gmail.com>

adamsitnik reviewed (approved):
LGTM, thank you @hangy !

hangy commented:
> @hangy thank you for your contribution! Please take a look at my comments.

Thanks for your input! Can't believe I actually missed to pass the append parameter. The test methods do look more concise, too. Hope I can provide better PRs if I get more familiar with the code base. 😃

adamsitnik commented:
> Thanks for your input! Can't believe I actually missed to pass the append parameter. The test methods do look more concise, too.

Don't worry about it! I've been working on the .NET Team for almost 5 years now and I usually get at least ten comments for even simplest PRs I create.

> Hope I can provide better PRs if I get more familiar with the code base. 

Looking forward to more PRs from you!
